ws-nextjs-app/utilities/cspHeaderResponse/index.ts

```diff
@@ -0,0 +1,72 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { helmetCsp } from '#server/utilities/cspHeader/directives';
+import { Services } from '#app/models/types/global';
+
+const directiveToString = (directives: Record<string, string | string[]>) => {
+  const map = new Map(Object.entries(directives));
+  let cspValue = '';
+  map.forEach((allowList, directive) => {
+    const allowListString = [allowList]
+      .flat()
+      .reduce((curr, acc) => `${curr} ${acc}`, '')
+      .replace(/\s{2,}/g, ' ');
+    cspValue += `${directive}${allowListString.length > 0 ? ' ' : ''}${allowListString};`;
+  });
+  return cspValue;
+};
+
+const cspHeaderResponse = ({
+  request,
+  isAmp = false,
+  isLive = false,
+  reportOnlyOnLive = false,
+  service,
+}: {
+  request: NextRequest;
+  isAmp?: boolean;
+  isLive?: boolean;
+  reportOnlyOnLive?: boolean;
+  service: Services;
+}) => {
+  const requestHeaders = new Headers(request.headers);
+
+  const helmetCSP = helmetCsp({
+    isAmp,
+    isLive,
+    reportOnlyOnLive,
+    service,
+  });
+
+  const nextJSSpecificHeaders = {
+    'base-uri': 'self',
+    'form-action': 'self',
+    'frame-ancestors': 'none',
+    'object-src': 'none',
+  };
```

Comment:
Not sure we need these. Its worth just trying the existing directives we have in the Express app and seeing that that list works.